core/src/main/java/de/danoeh/antennapod/core/feed/FeedItem.java

```diff
@@ -360,20 +328,25 @@ private boolean isPlaying() {
 
     @Override
     public Callable<String> loadShownotes() {
-        return () -> {
-            if (contentEncoded == null || description == null) {
-                DBReader.loadDescriptionOfFeedItem(FeedItem.this);
-            }
-            if (TextUtils.isEmpty(contentEncoded)) {
-                return description;
-            } else if (TextUtils.isEmpty(description)) {
-                return contentEncoded;
-            } else if (description.length() > 1.25 * contentEncoded.length()) {
-                return description;
-            } else {
-                return contentEncoded;
-            }
-        };
+        //doesn't process notes, ShownotesLoader is responsible for loading from database
+        //don't use this method
+        return () -> "";
+    }
+
+    public String getShownotes() {
```

Comment:
There are only very few cases in which we need to actually read the shownotes. How about removing this method completely and doing the checks in DbReader instead?

Reply:
Read above. I moved logic to completely new class. The idea behind it is that in future there might be more elements with shownotes, maybe they will be fetched through network, or read from file, or DB. If we put it in DbReader in future it might lead to scattered shownotes loading logic across different classes and packages.